### PR TITLE
RWIB Mission 3 Contributing Page

### PR DESCRIPTION
**Pull request by efindlay**
This is an adaptation of [ncamerons](http://www.ncameron.org/rust.html) page.

Todo: 
- It should have a link to instructions on submitting bug reports.
- It should lead people to instructions for getting involved in RFC process.
- It should provide leads for getting involved in evangelism.
- It might highlight major initiatives, like IDE's. Such initiatives would need
stable landing pages to link to, like internals threads or GitHub issues.

### CODE REVIEW

**Comment by brson** [comment]
Thanks @efindlay. I'll review Tuesday.

**Comment by brson** [comment]
Just dropping this link real quick: https://dxr.mozilla.org/rust/source/

DXR is supposedly pretty stable and quite useful, so we need to give it exposure.

**Comment by brson** [comment]
I've pushed PRS #201 (community), #200 (docs), and #199 (contributing) to [my own gh pages so reviewers can see](http://brson.github.io/rust-www/). The first goal is to get the community page merged since it's the most developed.

The key pages under review are [community.html](http://brson.github.io/rust-www/community.html), [user_groups.html](http://brson.github.io/rust-www/user_groups.html), [documentation.html](http://brson.github.io/rust-www/documentation.html), and [contributions.html](http://brson.github.io/rust-www/contributions.html)

**Comment by Manishearth** [comment]
IMO this should be somehow linked on the first page too.

File: contributions.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+layout: default
+title: How To Contribute
+---
+
+# Contributing to Rust
+
+If you're learning Rust, you've no doubt been amazed at how awesome
+the language is and will be keen to write more Rust code. Hopefully,
+you'll also want to make a meaningful contribution to the community.
+If you're not sure how best to do that, then hopefully this page will
```
Comment (Manishearth) [inline contributions.md]: nit: repeated hopefully.

File: contributions.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+[rust_issues]: https://github.com/rust-lang/rust/issues
+[e_easy_issues]: https://github.com/rust-lang/rust/issues?q=is%3Aopen+is%3Aissue+label%3AE-easy
+[e_mentor_issues]: https://github.com/rust-lang/rust/issues?q=is%3Aopen+is%3Aissue+label%3AE-easy+label%3AE-mentor
+
+#### The Rust compiler
+
+The compiler is part of the [main repo][main_repo], which also
+includes the standard library crates and a whole bunch of supporting
+code. For questions about the compiler, there is the
+[#rustc][rustc_irc] IRC channel. Compiler errors (ICE for 'internal
```
Comment (Manishearth) [inline contributions.md]: nit: parenthetical should be after the thing it's explaining (end of sentence)

**Comment by Manishearth** [comment]
This looks great to me. I may be rearranging the stuff in this page when I add the other community libraries, but for now it's pretty good.

**Comment by ashleygwilliams** [comment]
hi! i think this is a great start to a contributing guide- however, it only answers 1 question: "what are the types of things i could work on"

it doesn't answer questions like
- what needs most help
- what is good for a first pr
- how/what i need to do to contribute to a specific thing (proj deps etc)

an example i think of is: i want to contribute to docs. mentioning in this that docs are included in the rust repo itself, that they versioned, how that is affected by release cycles, etc would be all be super useful as they are important things to consider when deciding what and how to contribute.

perhaps this info should be in separate `contributing.md`s? but i would certainly link to them from here.

**Comment by efindlay** [comment]
@ashleygwilliams Thanks for these good points. I'll make three new sections in contributing.md for each of these questions and get working on it.

**Comment by brson** [comment]
I've posted a [counter-PR](https://github.com/efindlay/rust-www/pull/4) against @efindlay's branch. It doesn't do a lot new, but it does shuffle bits around: organizes the page into six sections that are listed up front with in-page links.

I was hoping this would be simplifying, but my rewritten section on 'bugs' is pretty sprawling. I'll give closer attention to all the other individual sections tomorrow.

**Comment by brson** [comment]
@efindlay I've continued hacking at your version [here](https://github.com/brson/rust-www/blob/contributions/contributions.md). I've really gone crazy with it and overhauled the content from top to bottom to be more comprehensive and explanatory. Sorry for just jumping in and making the changes myself but it felt easier than trying to go back and forth on it.

**Comment by efindlay** [comment]
No problem!

**Pull request closed without merging**